### PR TITLE
Fix #783 duplicate uploading same asset name

### LINKED ISSUE #783: Duplicate uploading same asset name
**- This is a *bug*?**

**- What is the current behavior?**

**- Reproduce.**
Open the media library ui and upload the same asset. A duplicate will show as in the picture above.

**- What is the expected behavior?**
Should only show one thumbnail per item

**- Please mention your CMS, node.js, and operating system version.**
Chrome Browser, et. al.

### PR DESCRIPTION
**Pull request by Dammmien**
# #783 duplicate uploading same asset name

**- Summary**

Fixes #783 Duplicate uploading same asset name

**- Test plan**

Upload an image with the same name, it will ask you to replace the existing one.

**- Description for the changelog**

Juste check existing files name

### CODE REVIEW

**Comment by erquhart** [comment]
Thanks for digging into this, @Dammmien, definitely something the CMS needs. So there's a bit of complexity to consider:

1. Some CMS implementations utilize a custom asset storage backend that returns paginated results, which the media library accesses via infinite scroll. That means we won't always have all of the files on hand to compare.

2. If assets are stored in Git, you can expect unique filenames, but not all assets will be stored in Git. In such cases, filenames aren't guaranteed to be unique, and an id is used instead.

We refer to the non-Git asset stores as "integrations", and your fix would need to be applied only in the absence of an asset integration. We also probably want this functionality in a Redux action rather than in the component itself.

Here's the associated action:
https://github.com/netlify/netlify-cms/blob/master/src/actions/mediaLibrary.js#L100-L138

You'll notice a bunch of it is conditionally run if there's an integration, and the rest runs if not - that last part is where you'll want to run your check and throw the prompt. I know GitHub won't paginate file results, and we can worry about other backends (e.g. GitLab/Bitbucket) as they get merged in.

If you can make that change, we should be in good shape to merge. Thanks again for your work!

**Comment by Dammmien** [comment]
@erquhart, not sure to understand, something like the last commit ?

File: src/actions/mediaLibrary.js [hunk 1]
```diff
@@ -80,6 +80,15 @@ export function persistMedia(file, opts = {}) {
       const assetProxy = await createAssetProxy(file.name.toLowerCase(), file, false, privateUpload);
       dispatch(addAsset(assetProxy));
       if (!integration) {
+        if (existingFile) {
+          if (!window.confirm('This media already exist, do you want to replace it?')) {
+            return dispatch(mediaPersistFailed({ privateUpload }));
+          }
+
+          const deletedFile = await deleteMedia(existingFile, { privateUpload });
```
Comment (erquhart) [inline src/actions/mediaLibrary.js]: `deleteMedia` returns a function, why not just dispatch it?

**Comment by erquhart** [comment]
Yep, and I linked to the wrong action, but I see that didn't throw you off :)

Just one comment in my review and this should be good!

**Comment by erquhart** [comment]
Hmm no deploy preview for some reason. Going to sanity check it locally before approving, but looks good - thanks @Dammmien!

**Comment by erquhart** [comment]
@Dammmien I pushed an update to get all existing file logic into the action. There's still an outstanding issue where the last file uploaded in the current session doesn't get caught in the check, we should get that straightened before merging.

**Comment by verythorough** [comment]
Deploy preview ready!

Built with commit dc6807d07a176a79bbc93d2f5c01277fcc0ef8b6

https://deploy-preview-853--netlify-cms-www.netlify.com

**Comment by erquhart** [comment]
Note: we should also fix #540 here by making the check case-insensitive.

**Comment by verythorough** [comment]
Deploy preview ready!

Built with commit 5f683007475d77f022f1095cda21bc8dfecd0a87

https://deploy-preview-853--cms-demo.netlify.com

**Comment by Dammmien** [comment]
@erquhart I pushed an update to compare file name and existing files names in lower case to make the check case-insensitive.

**Pull request merged**